Documentation/ApiOverview/Environment/ApplicationContext.rst

```diff
@@ -0,0 +1,340 @@
+:orphan:
+
+..  include:: /Includes.rst.txt
+..  index:: ! Application Context
+..  _application-context:
+
+===================
+Application Context
+===================
+
+In previous TYPO3 versions, error messages were displayed unfiltered in the
+frontend and often revealed security-relevant data such as database access,
+e-mail addresses, SMTP access and similar to the public. The so-called
+`ApplicationContext` was introduced to prevent this. These contexts can
+pre-configure TYPO3 to a certain extent and can also be queried again at
+various points such as the site configuration, e.g. to provoke a different
+:ref:`base variants <sitehandling-baseVariants>`.
+
+TYPO3 is delivered with 3 different modes, which affect the behaviour of TYPO3
+as follows:
+
+*   Production
+
+    *   Only errors are logged.
+    *   There is no logging of obsolete (deprecated) function calls.
+    *   In the event of an error, the frontend only displays:
+        `Oops, an error occurred! Code: {code}`.
+    *   The :ref:`Dependency injection <Dependency-Injection>` cache can only
+        be emptied using the Flush Cache button in the Installtool.
+    *   Calling up the Installtool menu items requires the additional entry
+        of a password.
+    *   Only admins with system maintainer authorisation can see the
+        Installtool menu items in the TYPO3 backend.
+    *   This mode is the best performing and most secure.
+
+*   Development
+
+    *   Errors and warnings are logged.
+    *   Obsolete (deprecated) function calls are logged in an additional
+        log file.
+    *   The error appears in the frontend with a note on where and in which
+        file it occurred.
+    *   In the backend, the corresponding table field is also displayed in
+        square brackets after each label when editing data records.
+    *   The :guilabel:`Clear All Cache` button at the top right also clears
+        the :ref:`Dependency injection <Dependency-Injection>` cache.
+    *   The menu items in the backend for the install tool no longer require an
+        additional password entry.
+    *   Admins without system maintainer authorisation can also see the menu
+        items for the install tool.
+
+*   Testing
+
+    *   In this special mode, caching
+        for :ref:`Class Loading <composer-class-loader>` is switched off or is
+        only valid for one request.
+
+Default ApplicationContext
```

Comment:
Please add an anchor to any header you introduce so it can be easily linked

Reply:
Done